internal/controller/controlplane_controller.go

```diff
@@ -145,75 +146,32 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
-	// FIXED: Process PostCreateHooks INDEPENDENTLY of everything else
-	// This breaks the chicken-egg cycle!
-	if hcp.Spec.PostCreateHook != nil || len(hcp.Spec.PostCreateHooks) > 0 {
-		log.Info("Processing PostCreateHooks independently")
-
-		// Get the appropriate reconciler to process hooks
-		var baseReconciler interface {
-			ReconcileUpdatePostCreateHook(ctx context.Context, hcp *tenancyv1alpha1.ControlPlane) error
-		}
-
-		switch hcp.Spec.Type {
-		case tenancyv1alpha1.ControlPlaneTypeK8S:
-			reconciler := k8s.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
-			baseReconciler = reconciler
-		case tenancyv1alpha1.ControlPlaneTypeOCM:
-			reconciler := ocm.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
-			baseReconciler = reconciler
-		case tenancyv1alpha1.ControlPlaneTypeVCluster:
-			reconciler := vcluster.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
-			baseReconciler = reconciler
-		case tenancyv1alpha1.ControlPlaneTypeHost:
-			reconciler := host.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
-			baseReconciler = reconciler
-		case tenancyv1alpha1.ControlPlaneTypeExternal:
-			reconciler := external.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
-			baseReconciler = reconciler
-		}
-
-		if baseReconciler != nil {
-			log.Info("Processing PostCreateHooks", "controlPlane", hcp.Name)
-			if err := baseReconciler.ReconcileUpdatePostCreateHook(ctx, hcp); err != nil {
-				log.Error(err, "Failed to process PostCreateHooks")
-				// Don't return error, just log it and continue
-			}
-			// After processing hooks, refresh the hcp object to get updated status
-			if err := r.Get(ctx, client.ObjectKey{Name: hcp.Name}, hcp); err != nil {
-				log.Error(err, "Failed to refresh ControlPlane after hook processing")
-			}
-		}
-	}
-
-	// ALWAYS call the type-specific reconciler to handle infrastructure (API server, etc.)
-	log.Info("Calling type-specific reconciler for infrastructure setup", "type", hcp.Spec.Type)
+	// PHASE 1: Type-specific controlplane reconciliation
+	log.Info("Phase 1: Infrastructure setup and kubeconfig creation", "type", hcp.Spec.Type)
 	var reconcileResult ctrl.Result
 	var reconcileError error
```

Comment:
FYI, these variable declarations could be simply deleted if the one statement that assigns to these variables used `:=` rather than `=`.